pkg/profile/artifact.go

```diff
@@ -7,54 +7,181 @@ import (
 	}
 
-	switch kind := p.definition.Spec.Artifacts[0].Kind; kind {
-	case "HelmChart":
-		if err := p.createHelmRelease(ctx); err != nil {
+	if helmRes != nil {
+		p.log.Info("creating HelmRelease", "resource", helmRes.ObjectMeta.Name)
+		if err := p.client.Create(ctx, helmRes); err != nil {
 			return fmt.Errorf("failed to create HelmRelease resource: %w", err)
 		}
-	case "Kustomize":
-		if err := p.createKustomization(ctx); err != nil {
+	} else {
+		p.log.Info("creating Kustomization", "resource", kustomizeRes.ObjectMeta.Name)
+		if err := p.client.Create(ctx, kustomizeRes); err != nil {
 			return fmt.Errorf("failed to create Kustomization resource: %w", err)
 		}
-	default:
-		return fmt.Errorf("artifact kind %q not recognized", kind)
 	}
-
 	p.log.Info("all artifacts created")
 	return nil
 }
 
+// ArtifactStatus checks if the artifacts exists and returns any ready!=true conditions on the artifacts.
+func (p *Profile) ArtifactStatus(ctx context.Context) (Status, error) {
+	resourcesExist, gitRes, helmRes, kustomizeRes, err := p.getResources(ctx)
+	if err != nil {
+		return Status{}, err
+	}
+
+	return Status{
+		ResourcesExist:     resourcesExist,
+		NotReadyConditions: p.checkResourcesReady(gitRes, helmRes, kustomizeRes),
+	}, nil
+}
+
+func (p *Profile) checkResourcesReady(gitRes *sourcev1.GitRepository, helmRes *helmv2.HelmRelease, kustomizeRes *kustomizev1.Kustomization) []metav1.Condition {
+	var notReadyConditions []metav1.Condition
+	if gitRes != nil {
+		if condition := getNotReadyCondition(gitRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+
+	if helmRes != nil {
+		if condition := getNotReadyCondition(helmRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+
+	if kustomizeRes != nil {
+		if condition := getNotReadyCondition(kustomizeRes.Status.Conditions); condition != (metav1.Condition{}) {
+			notReadyConditions = append(notReadyConditions, condition)
+		}
+	}
+	return notReadyConditions
+}
+
+func (p *Profile) getResources(ctx context.Context) (bool, *sourcev1.GitRepository, *helmv2.HelmRelease, *kustomizev1.Kustomization, error) {
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+	if err != nil {
+		return false, nil, nil, nil, err
+	}
+	gitResExists := true
+	helmResExists := true
+	kustomizeResExists := true
+	if gitRes != nil {
+		gitResExists, err = p.getResourceIfExists(ctx, gitRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if helmRes != nil {
+		helmResExists, err = p.getResourceIfExists(ctx, helmRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+
+	if kustomizeRes != nil {
+		kustomizeResExists, err = p.getResourceIfExists(ctx, kustomizeRes)
+		if err != nil {
+			return false, nil, nil, nil, err
+		}
+	}
+	return gitResExists && helmResExists && kustomizeResExists, gitRes, helmRes, kustomizeRes, nil
```

Comment:
this is one hell of a return signature 😂

Reply:
It's that or multiple returns for each Boolean :/

Reply:
Do you need the booleans?

I think you can simplify this by returning nil if the object isn't found? (which you're doing).

Reply:
I think its just simpler to keep the logic around if the resources existed inside the func. If no boolean is returned and just nil objects, how do we know `kustomizeRes` is nil because there isn't a kustomize artifact in the profile definition or because it was missing in the cluster?

Reply:
In the multiple artifacts PR I've changed it to this https://github.com/weaveworks/profiles/pull/61/files#diff-b425bde9d0ac4c1a499461b4ef45b4d9983833ab9025dccb91e85ac893df3a45R93-R118
```
	for _, helmRes := range helmResources {
		helmResExists, err := p.getResourceIfExists(ctx, helmRes)
		if err != nil || !helmResExists {
			return false, nil, nil, nil, err
		}
	}

	for _, kustomizeRes := range kustomizeResources {
		kustomizeResExists, err := p.getResourceIfExists(ctx, kustomizeRes)
		if err != nil || !kustomizeResExists {
			return false, nil, nil, nil, err
		}
	}
	return true, gitRes, helmResources, kustomizeResources, nil